### PR TITLE
Add more overview and introduction text : continuation

### PR DESCRIPTION
**Pull request by astrale-sharp**
this is a continuation of PR #24 , with sugggestion and comments applied.

I am not a native english speaker so fill free to point out weird formulations, typos and such ;)

### CODE REVIEW

File: src/gdnative-overview.md [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+2. `Resource` types such as `Mesh`, `Texture` or `PackedScene` 
+3. Data you can send to or receive from other scripts (perhaps in GDScript) such as `Variant` (eg: values, dictionaries or lists)  
+4. How to create your own types, supporting incoming calls.
+
 Subchapters:
 
 1. [Data representations](gdnative-overview/data-representations.md)
-1. [`Ref`, `TRef` and `Instance`](gdnative-overview/wrappers.md)
-1. [Game architecture](gdnative-overview/architecture.md)
+1. [`Ref`, `TRef` and `Instance`](gdnative-overview/wrappers.md)
```
Comment (Bromeon) [inline src/gdnative-overview.md:16]: _Game architecture_ shouldn't be removed.

File: src/gdnative-overview.md [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+GDNative is the interface between the Godot engine and bindings in native languages, such as C, C++ or Rust. It defines how Godot finds and loads dynamic libraries (DLLs / shared objects / dylibs) and understands its contents. 
 
 This chapter gives a broad overview of basic GDNative concepts and godot-rust's approach to implement them in Rust. It is not a usage guide for exposing your Rust code to Godot; see chapter [Binding to Rust code](rust-binding.md) for concrete examples.
 
+When working with GDNative you'll be primarily worried about:
+1. `Node` types such as `Node` or `Spatial`  
+2. `Resource` types such as `Mesh`, `Texture` or `PackedScene` 
+3. Data you can send to or receive from other scripts (perhaps in GDScript) such as `Variant` (eg: values, dictionaries or lists)  
+4. How to create your own types, supporting incoming calls.
+
```
Comment (Bromeon) [inline src/gdnative-overview.md:12]: This is a bit unstructured. Different data types are already mentioned [here](https://godot-rust.github.io/book/gdnative-overview/data-representations.html), and categorized more precisely. I would remove this section and focus more on the one below ("What is godot-rust")

File: src/introduction.md [hunk 3]
```diff
@@ -6,10 +6,24 @@ If you're new to `godot-rust`, try the [Getting Started](./getting-started.md) t
 
 If you have used earlier versions of `godot-rust` before, see [Migrating from godot-rust 0.8.x](./migrating-0-8.md) for a quick guided tour to the new API.
 
+## What is godot-rust
```
Comment (Bromeon) [inline src/introduction.md:9]: Question mark 🙂 
or: "What godot-rust is"

File: src/introduction.md [hunk 4]
```diff
@@ -6,10 +6,24 @@ If you're new to `godot-rust`, try the [Getting Started](./getting-started.md) t
...
+
+The Godot Engine has rich user scripting support. Using a DLL binding system called GDNative you can create "Scripts" in Rust and attach them to Godot Nodes. Everything you can do from Godot in a language such as GDScript you can do from rust using godot-rust.
+
+Your code is compiled into a dylib (often called a shared library or DLL) which is loaded by Godot and also has full access to Rust's entire feature set. You can read and write files, open network ports, connect to a custom server, run a custom website and more.
+
+Rust itself is a far more rigorous and complete language than GDScript, offering high level features such as traits or easy packing and unpacking of values. Rust offers very good performance that is comparable with C and C++, by designing most of them as zero-cost abstractions. so if your game simulation is complex, rust is a good language to run it.
+
+Once you have built your godot-rust project, you explain to Godot where to find the DLL and name each script, each using a little Godot resource file. Godot will open the compiled DLL and execute it when your game runs. See [Getting Started](./getting-started.md) to get an idea how this works.
```
Comment (Bromeon) [inline src/introduction.md:17]: Lot of things:
* capitalization: "Rust", "node"
* replace "DLL" with "shared library", it's not about Windows only.
* "Your code is compiled into a dylib (often called a shared library or DLL)"
   This is wrong, it's a `cdylib`, and officially named "dynamic system library" (.dll, .so, .dylib). Maybe you can link to https://doc.rust-lang.org/reference/linkage.html.
* "Everything you can do from Godot in a language such as GDScript you can do from rust using godot-rust."
   Also wrong, quite a few features are missing in Rust. I would omit this sentence, it promises things we don't have.

File: src/introduction.md [hunk 5]
```diff
@@ -6,10 +6,24 @@ If you're new to `godot-rust`, try the [Getting Started](./getting-started.md) t
...
 
 ## License
 
-The `godot-rust` bindings, and this user guide, are licensed under the [MIT license](LICENSE.md).
\ No newline at end of file
+The `godot-rust` bindings, and this user guide, are licensed under the [MIT license](LICENSE.md).
+
+## Help
+
+There are several active discords which discuss Godot-Rust. [Here is an invite to the godot-rust one](https://discord.gg/AbjQp6mjKS).
```
Comment (Bromeon) [inline src/introduction.md:29]: Already covered in https://godot-rust.github.io/book/faq/community.html, no need to mention here, too.

**Review by Bromeon — CHANGES_REQUESTED** [review]
Thank you! Now that I look at the old text again, I see there's a lot of things that need changing, and some sections have already been added in other parts of the book. So, I'm not really sure if we truly add value by just fixing the issues there; we need to add quite a bit of new content. I added remarks in the code for the understanding, but honestly I think rewriting makes more sense than fixing. Sorry for the confusion.

Concretely, the PR doesn't cover the things discussed in https://github.com/godot-rust/godot-rust/issues/770 -- while there is a mention of dynamic linking, it's very abstract and does not guide the user in how to do so. The interesting parts are setting up the .gdnlib and .gdns files, registering it with Godot and seeing the changes in the editor.

[This is a good introduction](https://hagsteel.com/posts/godot-rust/), for example. But please don't copy-paste other people's work without crediting them. Either we do it anew, or we limit this PR to placing an external link (which is also fine by me).

**Comment by jacobsky** [comment]
Hi! Thanks for the PR. One thing that I noticed is that this appears to be mostly the same as the previous PR without any of the attributions. It doesn't seem to add anything else beyond the previous PR. In reviewing the previous PR, I think that it seems like most of the Previous PR is a bit more fluff that is covered in other sections like the FAQ.

Since the PR is currently taking most of the previous with only some fixes to grammar and punctuation, to properly attribute everyone's work, I'd prefer if we try to keep all of the relevant commits, such as by placing any changes on top of the previous PR (this would involve pulling from the PR branch and working from there).

I've also been going over it, and I think that the big point is adding some quick links to the FAQ from the intro section to help guide users into finding the information. So I'll be working on that separately, just FYI.

**Comment by astrale-sharp** [comment]
Hi! I think i was a bit confused by your previous comment @Bromeon , I dont really see what i could personally add to the overview / introduction. I thought you just wanted a bit of refreshment of the PR (also, unrelated but I'm unsure about how to pull from the pull request that was already there)

I was indeed more interested in writing an explanation on how to implement what we discussed in #770 , rather than continuing this!
I'll close this and open another draft about that other topic if that's okay!

**Comment by Bromeon** [comment]
>  I'll close this and open another draft about that other topic if that's okay!

Maybe coordinate this with @jacobsky first, not that you both end up writing the same thing.

**Pull request closed without merging**